idea/idea-frontend-fir/idea-fir-low-level-api/src/org/jetbrains/kotlin/idea/fir/low/level/api/lazy/resolve/FirLazyDeclarationResolver.kt

```diff
@@ -56,7 +56,7 @@ internal class FirLazyDeclarationResolver(
         val firFile = declaration.getContainingFile()
             ?: error("FirFile was not found for\n${declaration.render()}")
         val provider = firFile.session.firIdeProvider
-        val fromPhase = if (reresolveFile) declaration.resolvePhase else minOf(firFile.resolvePhase, declaration.resolvePhase)
+        fun fromPhase() = if (reresolveFile) declaration.resolvePhase else minOf(firFile.resolvePhase, declaration.resolvePhase)
```

Comment:
I'm trying to understand how this change could resolve the race condition, and it seems to me, the timing of evaluating `fromPhase` matters. Previously, it's computed before any of resolvers ran, and now with this change, it could be computed when it's needed. Does this sound like `lazy`? How about this?

```kt
val fromPhase by lazy {
    if (reresolveFile) declaration.resolvePhase else minOf(firFile.resolvePhase, declaration.resolvePhase)
}
```

Reply:
Sure either way works for me.